src/analysis.zig

```diff
@@ -2071,6 +2071,28 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
         .greater_than,
         .less_or_equal,
         .greater_or_equal,
+        => {
+            const ty = try analyser.resolveTypeOfNodeInternal(
+                .{ .node = datas[node].lhs, .handle = handle },
+            ) orelse return null;
```

Comment:
> I tried solving this but I'm not sure if this is the right way to do it, here's some thoughts I had:
> 
>     * also check `rhs`
> 
>     * maybe instead of `orelse return null;` return `.bool_type` like before

Falling back to returning bool seems reasonable.
```suggestion
            ) orelse return try Type.typeValFromIP(analyser, .bool_type);
```
I think we can skip checking the `rhs`, the most common cases should already be handled without it and we save some time that would be spent on analyzing the extra expression.